flask_resty/jwt.py

```diff
@@ -70,8 +70,25 @@ def get_token_from_authorization(self, authorization):
     def get_token_from_request(self):
         return flask.request.args.get(self.id_token_arg)
 
+    def get_key_for_token(self, keys, token):
+        unverified_header = jwt.get_unverified_header(token)
+
+        try:
+            token_kid = unverified_header['kid']
+        except KeyError:
+            raise InvalidTokenError('Key ID header parameter is missing')
```

Comment:
I would do `raise ApiError(401, {'code': 'invalid_token.missing_kid'})` or something like that otherwise it would be hard to debug

Reply:
i wasn't sure if the lack of specific error generally around `decode` was intentional to not leak details?

Reply:
imo instead of doing this, we should have a separate `JwksAuthentication` that gets configured with the JWKS

Reply:
i'm not sure how this comment got carried over, but `invalid_token.missing_kid` sounds good

Reply:
oh or does this get handled upstream?